Competitive_Programming/Python/Graphs/Floyd-Warshall Algorithm/readme.md

```diff
@@ -0,0 +1,66 @@
+# Floyd-Warshall Algorithm
+
+<b>Floyd-Warshall Algorithm</b> is an algorithm for finding the shortest path between all the pairs of vertices in a weighted graph. 
+
+## Algorithm:
+
+```
+n = no of vertices
+A = matrix of dimension n*n
+for k = 1 to n
+    for i = 1 to n
+        for j = 1 to n
+            Ak[i, j] = min (Ak-1[i, j], Ak-1[i, k] + Ak-1[k, j])
+return A
+```
+
+## Python Code:
+
+```python
+nV = 4
+
+INF = 999
+
+def floyd_warshall(G):
+    distance = list(map(lambda i: list(map(lambda j: j, i)), G))
+
+    for k in range(nV):
+        for i in range(nV):
+            for j in range(nV):
+                distance[i][j] = min(distance[i][j], distance[i][k] + distance[k][j])
+    print_solution(distance)
+
+def print_solution(distance):
+    for i in range(nV):
+        for j in range(nV):
+            if(distance[i][j] == INF):
+                print("INF", end=" ")
+            else:
+                print(distance[i][j], end="  ")
+        print(" ")
+
+G = [[0, 3, INF, 5],
+         [2, 0, INF, 4],
+         [INF, 1, 0, INF],
+         [INF, INF, 2, 0]]
+floyd_warshall(G)
+```
+
+## Output:
+
+![output](https://user-images.githubusercontent.com/65852362/141053638-428efa68-dbb5-4c9d-9c4e-e3652975a25c.png)
+
+## Time Complexity:
+
+There are three loops. Each loop has constant complexities. So, the time complexity of the <b>Floyd-Warshall algorithm</b> is <b>O(n3)</b>.
```

Comment:
Is the complexity is n-cube...? @Jinal4502 
if yes, then you represent it like this n<sup>3</sup>

syntax: ```n<sup>3</sup>```

Reply:
sure, will do that